### ISSUE TITLE
[Feature] K8s 1.29 Support

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#43110]
Umbrella issue for supporting K8s 1.29 in Rancher.
 - [x] rancher/calico-cni
      - https://github.com/rancher/calico-cni/pull/10
 - [x] rancher/flannel-cni
      - https://github.com/rancher/flannel-cni/pull/12
      - CNI_PLUGIN_VERSION=v1.4.1
      - FLANNEL_VERSION=v1.4.0-flannel1
 - [x] rancher/image-mirror
      - https://github.com/rancher/image-mirror/pull/618
      - https://github.com/rancher/image-mirror/pull/624
      - Bump etcd to 3.5.12 
      - Bump coredns to 1.11.1 
      - Bump flannel to 0.25.1
      - Bump capi to v1.6.4
 - [x] rancher/rke-tools
      - https://github.com/rancher/rke-tools/pull/181
      - Bump cri-dockerd to v0.3.11
 - [x] rancher/provisioning
     - https://github.com/rancher/provisioning/pull/9
     - https://github.com/rancher/provisioning/pull/10
     - Bump capi v1.6.4
     - Bump rancher/kubectl v1.29.2 
 - [x] rancher/charts 
     - https://github.com/rancher/charts/pull/3901
 - [x] rancher/shell
     - https://github.com/rancher/shell/pull/231  
 - [x] rancher/kubernetes
    - https://github.com/rancher/kubernetes/pull/131
    - Add support for kubernetes 1.29.3
 - [x] rancher/client-go
    - https://github.com/rancher/client-go/pull/11
    -  TODOs
        - [x]  A new base branch needs to be created in rancher/client-go repo for release-1.29 using the upstream client go 0.29.3 tag, and then the base branch of this PR needs to be updated with the created branch.
 - [x] rancher/lasso
    - https://github.com/rancher/lasso/pull/62
    -  Add v1.29 support till March patches (0.29.3).
    - Bump controller runtime to 0.16.3
 - [x] rancher/wrangler
     - https://github.com/rancher/wrangler/pull/422
     - Bump k8s dependencies to 0.29.3 till march patches
     - Bump go version to 1.21
 - [x] rancher/norman
      - https://github.com/rancher/norman/pull/497
 - [x] rancher/dynamiclistener
      - https://github.com/rancher/dynamiclistener/pull/107
 - [x] rancher/machine
     - https://github.com/rancher/machine/pull/247
 - [x] rancher/apiserver
     -  https://github.com/rancher/apiserver/pull/77  
 - [x] rancher/steve
     - https://github.com/rancher/steve/pull/207
 - [x] rancher/eks-operator
     - https://github.com/rancher/eks-operator/pull/496
 - [x] rancher/aks-operator
     -  https://github.com/rancher/aks-operator/pull/490
 - [x] rancher/gke-operator
     - https://github.com/rancher/gke-operator/pull/435
  - [x] rancher/kontainer-driver-metadata
     -  https://github.com/rancher/kontainer-driver-metadata/pull/1394  
 - [x] rancher/rke
     -  https://github.com/rancher/rke/pull/3575  
 - [x] rancher/rancher
     - https://github.com/rancher/rancher/pull/45380

Notes about breaking changes:
* [x] `cis-1.23` option is scheduled to be removed in favor of `cis` in RKE2 1.29 releases per https://github.com/rancher/rke2/blob/9e79849266360b1aa27fabe0d981bcc6bc069858/pkg/rke2/rke2.go#L274. There may be references to this in backend and certainly are in UI as it's set there. Heads-up @gaktive @nwmac 
* [ ] In-tree cloud providers will be disabled by default starting with 1.29 (see https://github.com/kubernetes/enhancements/issues/2395). Note that in 1.29 and until not-yet-announced minor version it will be possible to enable them by changing `DisableCloudProviders` to `false` but ideally we would like to work with default value of this setting so should probably plan for having a path forward for the remaining in-tree cloud providers (see https://github.com/rancher/rancher-docs/pull/844 for how it was tackled for AWS in-tree cloud provider removal in 1.27).
	* [ ] https://github.com/rancher/rancher/issues/44856
	* [ ] https://github.com/rancher/rancher/issues/44857
	* [x] https://github.com/rancher/rancher/issues/44858
	* [x] https://github.com/rancher/rancher/issues/44859
* [x] New supported go version for all the repos and pipeline is `>=1.21`.
* [x] Needs to removed PSP related changes from `integration/test` and `validation/test` scenarios.
* [ ]  Remove Deprecated PSP Code from [rancher/charts](https://github.com/rancher/charts) Repository (since we're only supporting Kubernetes versions `1.25` and above, and Pod Security Policy (PSP) is unsupported in versions beyond 1.25, we should PSP-related code from our charts repository. )

Additional areas:

- https://github.com/rancher/rancher/issues/45351
- https://github.com/rancher/rancher/issues/45350
- https://github.com/rancher/rancher/issues/45349
- https://github.com/rancher/rancher/issues/45348
- https://github.com/rancher/rancher/issues/45347
- https://github.com/rancher/qa-tasks/issues/1304
- https://github.com/rancher/rancher/issues/45621
- https://github.com/rancher/rancher/issues/45620

**Comment by pjbgf** [comment]
PSP types [were removed](https://github.com/kubernetes/api/commit/5a75c6e751ef86e3cc1bb758c2fc8c30df53b03c) on `k8s.io/api@v0.29.0` which will break [rancher/rancher](https://github.com/rancher/rancher/blob/release/v2.9/pkg/apis/management.cattle.io/v3/authz_types.go#L292) and [longhorn](https://sourcegraph.com/search?q=context:%40pjbgf/rancher+.PodSecurityPolicySpec+-path:vendor&patternType=standard&sm=1) once we bump to that version.

**Comment by krunalhinguu** [comment]
We will need to remove `psp.go` file from rke as well as rancher, which will break some of the apis in ui, which i have faced during my initial testing

**Comment by gaktive** [comment]
@krunalhinguu for the UI, is that the Ember (rancher/ui) or Vue (rancher/dashboard) where you're seeing an error? Do you have an example from your tests for reference so UI can proactively figure out what may need fixing?

**Comment by krunalhinguu** [comment]
https://zube.io/rancher/rancher/c/60170

Adding images for reference @gaktive 

![image (1) (1)](https://github.com/rancher/rancher/assets/25076781/445e614e-e4b8-4fdf-bb2c-54f18231c655)
![image](https://github.com/rancher/rancher/assets/25076781/88bdc8d0-572f-4a48-9213-5f4e0c32ad18)

**Comment by diogoasouza** [comment]
In the next release of our fork of helm, we shouldn't cherry-pick [this](https://github.com/rancher/helm/commit/367c8e2c5c2848f7d6c6486f3e0f90bd46169837) and [this](https://github.com/rancher/helm/commit/48208cfbee98f49d4563415d4a3f4635db29c469) commit. It was discussed with @kinarashah, and it's part of the effort to start using upstream helm instead of a fork of it.

**Comment by nwmac** [comment]
This is ember - we embed it in the dashboard UI @gaktive

**Comment by fernandomullerjr** [comment]
UP

**Comment by mitulshah-suse** [comment]
@gaktive Can you please confirm UI fix for PSPs is being handled in this issue https://github.com/rancher/dashboard/issues/9945
or should we open another issue for it? It will be a blocker for most of 1.29 related work.
Also, if that is the one, it will need to be backported to 2.8, since 1.29 is being targeted there.

**Comment by gaktive** [comment]
@mitulshah-suse for https://github.com/rancher/dashboard/issues/9945 as written, that's done and in QA for 2.9.0. I can't tell from this ticket if other changes to PSPs are required so let me know if we need another ticket to cover what else to do since we were based off another ticket: https://github.com/rancher/rancher/issues/39789. The only other ticket we have is more for `cis`: https://github.com/rancher/dashboard/issues/9971

I can file a back port for 9945 for 2.8.x.

**Comment by gaktive** [comment]
Backport to 2.8.x for UI tickets, so if there is a new epic backport for this, these can be linked:
- PSPs: https://github.com/rancher/dashboard/issues/10678
- CIS rename: https://github.com/rancher/dashboard/issues/10679

**Comment by snasovich** [comment]
This issue now tracks adding K8s 1.29 support into 2.9.x line of releases.

**Comment by Conall-airbyte** [comment]
might be an idea to specify that 1.29+ is not yet supported in the 'requirements' section of the installation docs, as I understand this might take some time to be available.

**Comment by diogoasouza** [comment]
Would it be possible to remove this [commit](https://github.com/rancher/helm/pull/48/commits/edd5deb0ec2db1f6d1ab6142bf5dd98f86785900) from this release? It was necessary because of wrong permissions, but this [steve PR](https://github.com/rancher/steve/pull/166) fixed that.

**Comment by nickvth** [comment]
When will 1.29 be available? Because at end of Q3 1.28.x will be EOL and we want to upgrade before that.

**Comment by gaktive** [comment]
From UI, https://github.com/rancher/dashboard/issues/9971 doesn't require any code changes so this should work once everything else lands.

**Comment by BloodyIron** [comment]
> From UI, [rancher/dashboard#9971](https://github.com/rancher/dashboard/issues/9971) doesn't require any code changes so this should work once everything else lands.

What version of Rancher is anticipated to have k8s 1.29 support?

**Comment by kinarashah** [comment]
v1.29 and v1.30 both will be available in Rancher v2.9.0, which is expected to be released around last week of July.

**Comment by vivek-shilimkar** [comment]
### Summary of K8s v1.29 support testing checklist.

- :stopwatch:  =   Not started
- :clock12:  =   In progress
- :no_entry:  =   Blocked
- :white_check_mark:  =  PASSED :white_check_mark: 
- :question: =  Have Question

- Local cluster checks - :white_check_mark:
  - Docker Install - :white_check_mark:
  - Docker install and Upgrade - :white_check_mark:
  - HA install -  :white_check_mark:
  - Local Cluster HA Upgrade - :white_check_mark: 

- RKE1 Validations
  - OS Certification -  :white_check_mark:
  - Hosted Clusters testing - :white_check_mark:

- KDM Testing - :white_check_mark: 
  - Fresh Install Checks -  :white_check_mark:
  - Upgrade checks -  :white_check_mark:
  - Canal + PNI enabled - :white_check_mark:
  - Downstream cluster ETCD Validation - :white_check_mark:  PASS  
  - Downstream Control Plane Validation - :white_check_mark:  PASS 

- ETCD Validations - :white_check_mark: 
- Control plane Validations - :white_check_mark: 

- Local cluster RKE1 HA upgrade -   :white_check_mark:
- Local cluster RKE2 HA upgrade -   :white_check_mark:
- Local cluster K3s HA upgrade -   :white_check_mark:
- RKE checks - :white_check_mark:
- Rancher CLI checks - :white_check_mark:
- RKE2 and K3s Checks - :white_check_mark: 

- Node Driver Cluster Provisioning - :white_check_mark: 
  - Downstream RKE2 cluster provisioning - :white_check_mark:
    - Fresh install :white_check_mark:
    - Upgrade :white_check_mark:
  - Downstream k3s cluster provisioning - :white_check_mark: 
    - Fresh install :white_check_mark:
    - Upgrade  :white_check_mark: 

- Windows RKE2 provisioning :white_check_mark:
- RKE2 Amazon Cloud provider enabled checks. - :white_check_mark: 
- Any other functional check for 1.27 - Service accounts - :clock12:

- Airgap checks - :white_check_mark:
  - RKE1 cluster - custom clusters - :white_check_mark:
  - RKE2 clusters - custom clusters - :white_check_mark:
  - K3s clusters - custom clusters - :white_check_mark:

- PSACT testing - :white_check_mark: (Markus)

- Conformance Checks - CNCF Conformance tests for RKE Kubernetes cluster
  - Rancher -  :white_check_mark: PR Merged
  - RKE - :white_check_mark:  PR Merged

**Comment by vivek-shilimkar** [comment]
Since OS checks are still remaining, moving tickets back to Actively QAing.

**Comment by felipewnp** [comment]
Sorry, I'm confused, why this issue was closed?

Is rancher v2.9.0 released?

It doesn't look so.

**Comment by ofekedri** [comment]
@felipewnp looks like it released (a bit after your comment)
[release 2.9.0](https://github.com/rancher/rancher/releases?q=2.9.0&expanded=true)

**Comment by PT-GD** [comment]
In case that link brought anyone else to the wrong release tag as it did for me, try https://github.com/rancher/rancher/releases/tag/v2.9.0 worked for me!

**Comment by kekedaine** [comment]
please support GKE 1.29.6
